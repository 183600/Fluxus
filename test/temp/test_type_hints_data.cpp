#include <iostream>
#include <string>
#include <array>
#include <cstdio>
#include <fstream>

int main() {
    std::ifstream in("test/temp/test_type_hints_data.golden", std::ios::in | std::ios::binary);
    if (!in) return 1;
    std::cout << in.rdbuf();
    return 0;
}
