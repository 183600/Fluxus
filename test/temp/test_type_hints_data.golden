=== Type Hints Test ===
Name: Alice, Age: 25
Scores: [90, 85, 95]
Person data: {'name': 'Bob', 'age': 30}
Optional name: Charlie
Mixed data: [1, 2, 3]

=== Generics Test ===
Int box: 42
Str box: Hello
Pair: (first, 2)
First item: 1
First item: a

=== Data Classes Test ===
Point 1: Point(x=3.5, y=4.2)
Point 2: Point(x=1.0, y=2.0)
Rectangle: Rectangle(width=5.0, height=3.0, position=Point(x=0, y=0), color='blue', area=15.0)
Rectangle area: 15.0
Immutable person: ImmutablePerson(name='Alice', age=25, email='alice@example.com')

=== Enums Test ===
Red color: Color.RED
Red value: #FF0000
Current status: Status.IN_PROGRESS
Status name: IN_PROGRESS
Status value: 2
All statuses:
  PENDING: 1
  IN_PROGRESS: 2
  COMPLETED: 3
  FAILED: 4

=== Protocols Test ===
Circle area: 78.53975
Drawing circle with radius 5.0

=== TypedDict Test ===
Person: {'name': 'Alice', 'age': 25, 'email': 'alice@example.com', 'active': True}
Person name: Alice
Person age: 25

=== Abstract Classes Test ===
Sound: Woof!
Movement: Running on four legs
Sound: Meow!
Movement: Walking gracefully

=== Advanced Types Test ===
Add: 8
Multiply: 10.0
Apply function result: 30
Squared numbers: [1, 4, 9, 16, 25]
Count of 2: 3

=== Literal Types Test ===
Set status: Status set to active
Set status: Status set to inactive
Max retries: 3
Default timeout: 30.0

=== Complex Types Test ===
Config: {'max_connections': 10, 'timeout': 30.5, 'debug_mode': True, 'allowed_hosts': ['localhost', '127.0.0.1']}
Processed data: ['HELLO', 'WORLD', 'TEST']

=== Union and Optional Types Test ===
Process int: Integer: 42
Process str: String: hello
Process None: None
Find 3: 2
Find 10: None

=== All type hints and data tests passed! ===